static/js/chart/draw.ts

```diff
@@ -404,10 +410,117 @@ function drawLineChart(
   }
 }
 
+/**
+ * Draw line chart.
+ * @param parentId
+ * @param parentWidth
+ * @param parentHeight
+ * @param dataGroups
+ * @param unit
+ */
+function drawLineChart(
+  parentId: string,
+  width: number,
+  height: number,
+  dataGroups: DataGroup[],
+  unit?: string
+) {
+  let parentElement = document.getElementById(parentId);
+  // Pick one group and use the data point label as text list.
+  let legendText = dataGroups.map((dataGroup) => dataGroup.label);
+  let minV = Math.min(...dataGroups.map((dataGroup) => dataGroup.min()));
+  let maxV = Math.max(...dataGroups.map((dataGroup) => dataGroup.max()));
+
+  // Draw legend and compute the height.
+  // Add legend to the container box in the end.
+  let hasLegend = dataGroups.length > 1;
+  let legendHeight = 0;
+  let legendElem;
+  if (hasLegend) {
+    legendElem = createLegend(legendText);
+    parentElement.appendChild(legendElem);
+    let bbox = legendElem.getBoundingClientRect();
+    legendHeight = bbox.height;
+    parentElement.removeChild(legendElem);
+  }
+  height -= legendHeight;
+  const margin = { top: 10, right: 10, bottom: 30, left: 30 };
```

Comment:
Could you pull this out to a const, since we will be sharing it across charts.

Reply:
Done